### PR TITLE
Longest Common Prefix in Java

### PR DESCRIPTION
**Pull request by rohangawhade**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added longest_common_prefix in java which aims to find the longest common prefix in all the entered strings

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#917

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rohangawhade Add time and space complexity. Resolve the merge conflicts.

**Comment by rohangawhade** [comment]
@iamrajiv I committed time and space complexity. It is showing one check failed. Can you please help me out with this. How to resolve this. Also in the README there's a file called 'Container with Most Water' with unknown location. There is no such file.

**Comment by iamrajiv** [comment]
@rohangawhade Do not worry about Prettier Action. Prettier has to be refactored.

File: Java/cp/Longest_Common_Prefix.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter strings");
+        String st = sc.nextLine();
+        String[] arrOfStr = st.split(" ");
+        Longest_Common_Prefix lcp = new Longest_Common_Prefix();
+        System.out.println("Longest Common Prefix : "+lcp.LongestCommonPrefixFn(arrOfStr));
+    }
+}
+
+//sample input and output:
```
Comment (iamrajiv) [inline Java/cp/Longest_Common_Prefix.java]: @rohangawhade Update all single-line comments into one multiline comment.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rohangawhade Update README like this
```md
<p align="center">
    <img src="../img/neo_algo.png"><br>
    <img src="https://img.shields.io/github/license/tesseractcoding/neoalgo?style=flat">
    <a href="http://makeapullrequest.com" target="_blank"><img src="https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat" alt="PRs Welcome"></a>
    <img alt="GitHub pull requests" src="https://img.shields.io/github/issues-pr/tesseractcoding/neoalgo">
    <img alt="GitHub issues" src="https://img.shields.io/github/issues/tesseractcoding/neoalgo">
    <img alt="Github All Contributors" src="https://img.shields.io/github/all-contributors/tesseractcoding/neoalgo">
</p>

# Codes in the Java language

## Algorithms

- [Kadane's Algorithm](Algorithms/kadanes_Java.java)

## Greedy Algorithms

- [Activity Selection](Greedy_Algorithms/Active_Selection.java)

## Blockchain

_add list here_

## Competitive programming

- [Aggressive Cows](cp/AggressiveCows.java)
- [Count The Reversals](cp/Count_The_Reversals.java)
- [Merge Overlaping Subintervals](cp/MergeOverlappingArray.java)
- [Rabin Karp Algorithm](cp/Rabin_Karp.java)
- [Remove Duplicate Element from an array using extra space](cp/RemoveDuplicateElement.java)
- [Remove Duplicate Element Without using Extra Space](cp/RemoveDuplicateElementWithoutExtraSpace.java)
- [Target Sum Triplets](cp/target_sum_triplets.java)
- [String to Int(atoi)](<cp/String_to_int(atoi).java>)

* [Aggressive Cows](cp/AggressiveCows.java)
* [Count The Reversals](cp/Count_The_Reversals.java)
* [Merge Overlaping Subintervals](cp/MergeOverlappingArray.java)
* [Rabin Karp Algorithm](cp/Rabin_Karp.java)
* [Remove Duplicate Element from an array using extra space](cp/RemoveDuplicateElement.java)
* [Remove Duplicate Element Without using Extra Space](cp/RemoveDuplicateElementWithoutExtraSpace.java)
* [Spiral Matrix](cp/Spiral_Matrix.java)
* [Target Sum Triplets](cp/target_sum_triplets.java)
* [Container with Most water](Container_with_most_area.java)
* [Remove Outer Most Parentheses](cp/Remove_Outermost_parentheses.java)

## Cryptography

- [Caesar Cipher](cryptography/ceaserCipher.java)

## Dynamic Programming

- [Coin Change Problem Using DP](dp/CoinChangeUsingDp.java)
- [Kandane Algorithm](dp/Kadane_Algorithm.java)

## Data Structures

- [Binary Tree](ds/Binary_Tree.java)
- [Detect Loop in Linked List](DetectLoopLL.java)
- [Matrix Sorting](ds/MatrixSorting.java) master
- [Queue Basic Functionalities](ds/queue.java)
- [Implementation of Queue using stack & vice-versa](ds/QueueandStack.java)
- [Queue](ds/Queuell.java)
- [Segment Tree](ds/SegmentTree.java)
- [Sparse Matrix](ds/SparseMatrix.java)
- [Stack ~ Linked List Implementation](ds/Stack.java)
- [Stack](ds/Stackll.java) Implementation-of-queue-using-stack
- [Linked List](ds/linkedList.java)
- [Linked List ~ Reverse](ds/linkedListReverse.java)
- [Tree Traversal(preorder traversal)](ds/preOrderTraversal.java)
- [Tree Traversal(postorder traversal)](ds/postorder_Traversal.java)

## Graphs

- [Breadth First Search](graphs/BFS.java)
- [Depth First Search](graphs/DFS.java)
- [Dijkstra's Algorithm](graphs/Dijkstra.java)
- [Kruskal's Algorithm](graphs/Kruskal_Algorithm.java)
- [Level Order Traversal of a Binary Tree Using Priority Queue](graphs/LevelOrderTraversalInQueue.java)
- [Prim's Algorithm](graphs/Prim_Algorithm.java)

## Mathematics

- [Count Number Of Digits](math/countDigits)
  - [Iterative](countDigits/iterative.java)
  - [Recursive](countDigits/recursive.java)
  - [String Method](countDigits/stringmethod.java)
- [Binary Exponentiation](math/Binary_Exponentiation.java)
- [Complex numbers](math/Complex.java)
- [Extended Euclid Algorithm](math/ExtendedEuclidAlgo.java)
- [Segmented_Sieve](math/Segmented_Sieve.java)
- [Sieve of Atkin](math/sieveOfAtkin.java)
- [Sieve of Eratosthenes](math/SieveOfEratosthenes.java)
- [Double Factorial](math/DoubleFactorial.java)

## Dynamic Programming

- [Kandane Algorithm](dp/Kadane_Algorithm.java)
- [Longest common subsequence](dp/LCS.java)

## Blockchain

_add list here_

## Searching

- [Binary Search](search/Binary_search.java)
- [Interpolation Search](search/Interpolation_search.java)
- [Linear Search](search/Linear_search.java)
- [Manacher's Algorithm](search/ManacherAlgorithm.java)
- [Stack Search](search/Stack_Search.java)
- [Jump Search](search/jumpSearch.java)

## Sorting

- [BeadSort](sort/BeadSort.java)
- [Bubble Sort](sort/BubbleSort.java)
- [Bucket Sort](sort/BucketSort.java)
- [Cocktail Sort](sort/CocktailSort.java)
- [Heap Sort](sort/HeapSort.java)
- [Insertion Sort](sort/InsertionSort.java)
- [Merge Sort](sort/Merge_sort.java)
- [Minimum number of swaps to sort an array of N unordered consecutive integers](sort/MinimumSwapsForNIntegers.java)
- [Optimized Bubble Sort](sort/OptimizedBubbleSort.java)
- [PancakeSort](sort/PancakeSort.java)
- [Quick Sort](sort/QuickSort.java)
- [Selection Sort](sort/SelectionSort.java)

## Competitive programming

- [Remove Duplicate Element from an array using extra space](cp/RemoveDuplicateElement.java)
- [Remove Duplicate Element Without using Extra Space](cp/RemoveDuplicateElementWithoutExtraSpace.java)
- [Rabin Karp Algorithm](cp/Rabin_Karp.java)
- [Merge Overlaping Subintervals](cp/MergeOverlappingArray.java)
- [Count The Reversals](cp/Count_The_Reversals.java)
- [Target Sum Triplets](cp/target_sum_triplets.java)
- [Aggressive Cows](cp/AggressiveCows.java)
- [Spiral Matrix](cp/Spiral_Matrix.java)
- [BeadSort](sort/BeadSort.java)
- [Bubble Sort](sort/BubbleSort.java)
- [Bucket Sort](sort/BucketSort.java)
- [Heap Sort](sort/HeapSort.java)
- [Insertion Sort](sort/InsertionSort.java)
- [Merge Sort](sort/Merge_sort.java)
- [Minimum number of swaps to sort an array of N unordered consecutive integers](sort/MinimumSwapsForNIntegers.java)
- [Optimized Bubble Sort](sort/OptimizedBubbleSort.java)
- [PancakeSort](sort/PancakeSort.java)
- [Quick Sort](sort/QuickSort.java)
- [Selection Sort](sort/SelectionSort.java)
- [Longest Substring without repeating characters](LongestSubstring_without_repeating_characters)
- [Longest Common Prefix](cp/Longest_Common_Prefix.java)

## Machine Learning

_add list here_

## Other

- [Find first missing positive number](other/FirstMissingPositiveNumber.java)
- [First non-repeating character in a string](other/FirstNonRepeatingChar.java)
- [Merge 2 Sorted arrays in O(1) extra space](other/MergeSortedArrays.java)
- [Minimum Sum Partition](other/MinimumSumPartition.java)
- [Tower of Hanoi problem](other/TowerOfHanoi.java)
```

File: Java/cp/Longest_Common_Prefix.java (hunk 1 above)
Comment (rohangawhade) [reply Java/cp/Longest_Common_Prefix.java]: @iamrajiv  Updated readme and the comments as suggested.

File: Java/README.md [hunk 2]
```diff
@@ -34,6 +34,19 @@ _add list here_
 - [Remove Duplicate Element from an array without using Extra Space](cp/RemoveDuplicateElementWithoutExtraSpace.java)
 - [Spiral Matrix](cp/Spiral_Matrix.java)
 - [String to Int(atoi)](<cp/String_to_int(atoi).java>)
+- [Longest Common Prefix](cp/Longest_Common_Prefix.java)
+
```
Comment (iamrajiv) [inline Java/README.md]: @rohangawhade remove this line break

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: Java/README.md (hunk 2 above)
Comment (rohangawhade) [reply Java/README.md]: @iamrajiv removed the line break

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rohangawhade In README `[Longest Common Prefix](cp/Longest_Common_Prefix.java)` is repeated twice. Also, I can see so much of edit in README just add `[Longest Common Prefix](cp/Longest_Common_Prefix.java)` do not edit other things.

**Comment by rohangawhade** [comment]
> @rohangawhade In README `[Longest Common Prefix](cp/Longest_Common_Prefix.java)` is repeated twice. Also, I can see so much of edit in README just add `[Longest Common Prefix](cp/Longest_Common_Prefix.java)` do not edit other things.

Now there is only one `[Longest Common Prefix](cp/Longest_Common_Prefix.java)` in the readme. @iamrajiv

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM :)

File: Java/cp/Longest_Common_Prefix.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import java.util.*;
+class Longest_Common_Prefix{
+    public String LongestCommonPrefixFn(String[] strs) {
```
Comment (vatsalkesarwani12) [inline Java/cp/Longest_Common_Prefix.java]: make this function static and call it directly from the main function

File: Java/cp/Longest_Common_Prefix.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            }
+        }
+        return st;
+    }
+    public static void main(String args[]){
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter strings");
+        String st = sc.nextLine();
+        String[] arrOfStr = st.split(" ");
+        Longest_Common_Prefix lcp = new Longest_Common_Prefix();
```
Comment (vatsalkesarwani12) [inline Java/cp/Longest_Common_Prefix.java]: make above changes and then there is no need to create object to call a function

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: Java/cp/Longest_Common_Prefix.java (hunk 4 above)
Comment (rohangawhade) [reply Java/cp/Longest_Common_Prefix.java]: @plazzy99  Made the function static.

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Comment by iamrajiv** [comment]
@all-contributors please add @rohangawhade for code

**Pull request merged**